util/collate/general_ci.go

```diff
@@ -24,30 +27,37 @@ type generalCICollator struct {
 func (gc *generalCICollator) Compare(a, b string) int {
 	a = truncateTailingSpace(a)
 	b = truncateTailingSpace(b)
-	r1, r2 := rune(0), rune(0)
-	ai, bi := 0, 0
-	for ai < len(a) && bi < len(b) {
-		r1, ai = decodeRune(a, ai)
```

Comment:
Well, is it possible to keep `decodeRune` with some additional validation checks?

Reply:
that will be the same with the golang library, it's faster because `decodeRune` has no validation check.